src/ucp/core/ucp_worker.c

```diff
@@ -1699,10 +1700,19 @@ static void ucp_worker_print_used_tls(const ucp_ep_config_key_t *key,
     p += strlen(p);
 
     for (lane = 0; lane < key->num_lanes; ++lane) {
-        if (((key->am_lane == lane) || (lane == key->tag_lane) ||
+        if (((key->am_lane == lane) ||
             (ucp_ep_config_get_multi_lane_prio(key->am_bw_lanes, lane) >= 0)  ||
-            (ucp_ep_config_get_multi_lane_prio(key->rma_bw_lanes, lane) >= 0)) &&
-            (context->config.features & UCP_FEATURE_TAG)) {
+            (ucp_ep_config_get_multi_lane_prio(key->rma_bw_lanes, lane) >= 0))) {
+            if (context->config.features & UCP_FEATURE_TAG) {
+                tag_lanes_map |= UCS_BIT(lane);
+            }
+
+            if (context->config.features & UCP_FEATURE_AM) {
+                am_lanes_map |= UCS_BIT(lane);
+            }
+        }
+
+        if (lane == key->tag_lane) {
```

Comment:
btw, why it is move here?

Reply:
because it is relevant for TAG feature only

Reply:
i.e. we should add it without checking TAG feature requested?

Reply:
we initialize tag lane if TAG feature is requested only, so checking for feature is not really needed. 
but i'll add the check just in case

Reply:
> we initialize tag lane if TAG feature is requested only, so checking for feature is not really needed.

I see

> but i'll add the check just in case

maybe worth to add an `assert()` instead?

Reply:
ok